processing/src/main/java/io/druid/segment/filter/AndFilter.java

```diff
@@ -181,6 +165,7 @@ public String toString()
 
   private ValueMatcher makeMatcher(final ValueMatcher[] baseMatchers)
   {
+    Preconditions.checkState(baseMatchers.length > 0);
```

Comment:
Can this support a 0 length array and just return `BooleanValueMatcher.of(false);`?